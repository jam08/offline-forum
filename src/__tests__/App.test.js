import React from 'react';
import { shallow, mount } from 'enzyme';
import sinon from  'sinon';
import App from '../components/App';
import * as api from '../api/index';

beforeEach(() => {
  localStorage.clear();
});

it('calls componentDidMount', () => {
  sinon.spy(App.prototype, 'componentDidMount');
  const wrapper = mount(<App />);
  expect(App.prototype.componentDidMount.calledOnce).toEqual(true);
  wrapper.unmount();
});

it('loads the app', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find('.mt-8').length).toBe(1);
});


it('loads comments');
it('no x button on other users posts');
it('input component takes users input');
it('should add new post when create button is clicked');
it('should add comment when comment button is clicked');
it('should delete post when x button is clicked');
it('loads bot page when talk to real human button is clicked');
it('get a reply from bot after sicka button is clicked');
it('sicka button should be disable if no message');