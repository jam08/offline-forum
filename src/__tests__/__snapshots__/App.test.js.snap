// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads <Bot /> if currentpage is bot 1`] = `
<div
  className="mt-8 mx-auto sm:w-full md:w-3/4 lg:w-2/3"
>
  <Button
    className="absolute pin-l pin-t mt-6 ml-2"
    danger={false}
    onClick={[Function]}
  >
    Return to forum
  </Button>
  <PersonaSwitcher
    changePersona={[Function]}
    currentPersona=""
  />
  <Bot />
</div>
`;

exports[`loads <Posts /> if current page is home 1`] = `
<div
  className="mt-8 mx-auto sm:w-full md:w-3/4 lg:w-2/3"
>
  <Button
    className="absolute pin-l pin-t mt-6 ml-2"
    danger={false}
    onClick={[Function]}
  >
    Talk to a real human
  </Button>
  <PersonaSwitcher
    changePersona={[Function]}
    currentPersona=""
  />
  <Posts
    currentPersona=""
  />
</div>
`;
